crates/topos-core/src/uci/subnet_id.rs

```diff
@@ -88,6 +88,16 @@ impl FromStr for SubnetId {
     }
 }
 
+impl PartialEq<[u8]> for SubnetId {
+    fn eq(&self, other: &[u8]) -> bool {
+        if let Ok(current) = Self::try_from(other) {
+            self.as_array().eq(current.as_array())
```

Comment:
What's the advantage of doing this? As opposed to `self.as_array() == other`? Why convert `other` to a SubnetId first?

Reply:
It validates the format of the subnet id provided as `[u8]` which can contains extra `0x` that need to be removed.